extension/tpcds/tpcds-extension.cpp

```diff
@@ -0,0 +1,198 @@
+                                                     vector<string> &input_table_names,
+                                                     vector<LogicalType> &return_types, vector<string> &names) {
+	names.emplace_back("query_nr");
+	return_types.push_back(LogicalType::INTEGER);
+
+	names.emplace_back("scale_factor");
+	return_types.push_back(LogicalType::INTEGER);
+
+	names.emplace_back("answer");
+	return_types.push_back(LogicalType::VARCHAR);
+
+	return nullptr;
+}
+
+static void TPCDSQueryAnswerFunction(ClientContext &context, const FunctionData *bind_data,
+                                     FunctionOperatorData *operator_state, DataChunk *input, DataChunk &output) {
+	auto &data = (TPCDSData &)*operator_state;
+	idx_t tpcds_queries = 22;
+	vector<double> scale_factors {0.01, 0.1, 1};
+	idx_t total_answers = tpcds_queries * scale_factors.size();
+	if (data.offset >= total_answers) {
+		// finished returning values
+		return;
+	}
+	idx_t chunk_count = 0;
+	while (data.offset < total_answers && chunk_count < STANDARD_VECTOR_SIZE) {
+		idx_t cur_query = data.offset % tpcds_queries;
+		idx_t cur_sf = data.offset / tpcds_queries;
+		auto answer = TPCDSExtension::GetAnswer(scale_factors[cur_sf], cur_query + 1);
+		// "query_nr", PhysicalType::INT32
+		output.SetValue(0, chunk_count, Value::INTEGER((int32_t)cur_query + 1));
+		// "scale_factor", PhysicalType::INT32
+		output.SetValue(1, chunk_count, Value::DOUBLE(scale_factors[cur_sf]));
+		// "query", PhysicalType::VARCHAR
+		output.SetValue(2, chunk_count, Value(answer));
+		data.offset++;
+		chunk_count++;
+	}
+	output.SetCardinality(chunk_count);
+}
+
+static string PragmaTpcdsQuery(ClientContext &context, const FunctionParameters &parameters) {
+	auto index = parameters.values[0].GetValue<int32_t>();
+	return tpcds::DSDGenWrapper::GetQuery(index);
+}
+
+void TPCDSExtension::Load(DuckDB &db) {
+	Connection con(db);
+	con.BeginTransaction();
+
+	TableFunction dsdgen_func("dsdgen", {}, DsdgenFunction, DsdgenBind);
+	dsdgen_func.named_parameters["sf"] = LogicalType::INTEGER;
+	// FIXME: no support for these parameters for now
+	//	dsdgen_func.named_parameters["overwrite"] = LogicalType::BOOLEAN;
+	//	dsdgen_func.named_parameters["schema"] = LogicalType::VARCHAR;
+	//	dsdgen_func.named_parameters["suffix"] = LogicalType::VARCHAR;
+	CreateTableFunctionInfo dsdgen_info(dsdgen_func);
+
+	// create the dsdgen function
+	auto &catalog = Catalog::GetCatalog(*con.context);
+	catalog.CreateTableFunction(*con.context, &dsdgen_info);
+
+	// create the TPCDS pragma that allows us to run the query
+	auto tpcds_func = PragmaFunction::PragmaCall("tpcds", PragmaTpcdsQuery, {LogicalType::BIGINT});
+	CreatePragmaFunctionInfo info(tpcds_func);
+	catalog.CreatePragmaFunction(*con.context, &info);
+
+	// create the TPCDS_QUERIES function that returns the query
+	TableFunction tpcds_query_func("tpcds_queries", {}, TPCDSQueryFunction, TPCDSQueryBind, TPCDSInit);
```

Comment:
Could you add a test for these functions?

Reply:
I added a call to `tpcds_queries` to the sf0 test, and commented out the answer function, since we do not support this yet.